velox/functions/prestosql/aggregates/PrestoHasher.cpp

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#define XXH_INLINE_ALL
+
+#include "velox/functions/prestosql/aggregates/PrestoHasher.h"
+#include "velox/external/xxhash.h"
+#include "velox/functions/lib/LambdaFunctionUtil.h"
+
+namespace facebook::velox::aggregate {
+
+namespace {
+
+template <typename T>
+FOLLY_ALWAYS_INLINE int64_t hashInteger(const T& value) {
+  int32_t distance = 31; // Making this a constant breaks hash calculation.
+  int64_t num = value * PRIME64_2;
+  int64_t unsignedShift = ((uint64_t)num) >> -distance;
+  int64_t rotateLeft = (num << distance) | unsignedShift;
+  return rotateLeft * PRIME64_1;
+}
+
+template <typename T>
+FOLLY_ALWAYS_INLINE int64_t
+hashInteger(const DecodedVector& vector, vector_size_t row) {
+  return vector.isNullAt(row) ? 0 : hashInteger(vector.valueAt<T>(row));
+}
+
+FOLLY_ALWAYS_INLINE int64_t
+hashStringView(const DecodedVector& vector, vector_size_t row) {
```

Comment:
I'd expect this function to be similar to hashInteger, but hashInteger checks for null while this function doesn't. Why is it not necessary for this function to check for null?